pkg/util/kubernetes/apiserver/leaderelection/leaderelection_test.go

```diff
@@ -27,3 +49,49 @@ func TestSuite(t *testing.T) {
 	s := &testSuite{}
 	suite.Run(t, s)
 }
+
+func TestStoppedLeading(t *testing.T) {
+	const leaseName = "datadog-leader-election"
+
+	client := fake.NewSimpleClientset()
+
+	lock := newFakeLockObject("default", leaseName, "foo")
+	_, err := client.CoreV1().ConfigMaps("default").Create(lock)
+	require.NoError(t, err)
+
+	le := &LeaderEngine{
+		HolderIdentity:  "foo",
+		LeaseName:       leaseName,
+		LeaderNamespace: "default",
+		LeaseDuration:   1 * time.Second,
+
+		coreClient: client.CoreV1(),
+	}
+
+	le.leaderElector, err = le.newElection()
+	require.NoError(t, err)
+
+	le.EnsureLeaderElectionRuns()
+
+	require.True(t, le.IsLeader())
+
+	lock = newFakeLockObject("default", leaseName, "bar")
+	_, err = client.CoreV1().ConfigMaps("default").Update(lock)
+	require.NoError(t, err)
+
+	// poll until election is lost
+	err = wait.Poll(1*time.Second, 10*time.Second, func() (done bool, err error) {
```

Comment:
Could we use `ResetGlobalLeaderEngine` to avoid waiting ?

Reply:
I need to wait here for the leader identity to transition. It should happen very fast since the `LeaseDuration` is only 1 second but there is no other mechanism for knowing when this occurs. 

The purpose of the test is to ensure that these transitions happen. 

Another solution is to wait on a channel that sends when the `OnStartedLeading`  callback is triggered. Would you prefer that?

Reply:
I did not realise that it was set at 1 second, so that should be fine as we have a 10 seconds timeout.